tutorials/persistance-comparison-series/Sqlite/sqlite.md

```diff
@@ -0,0 +1,388 @@
+
+All that's left to do is to properly `Close()` (12) the database.
+
+How can we actually verify that this worked out before we continue on to reading the values from the database again? Well, the easiest way would be to just look into the database. There are many tools out there to achieve this. One of the open source options would be [https://sqlitebrowser.org/](https://sqlitebrowser.org/).
+
+After downloading and installing it, all you need to do is `File -> Open Database`, and then browse to your Unity project and select the `MyDatabase.sqlite` file. If you then choose the `Table` `HitCountTableSimple`, the result should look something like this:
+
+<img src="images/07_db_browser.png" alt="Capsule in scene" width="800"/>
+
+Go ahead and run your game. Click a couple times on the capsule and check the Inspector for the change. When you then go back to the DB browser and click refresh, the same number should appear in the `value` column of the table.
+
+The next time we start the game, we want to load this hit count from the database again. We use the `Start()` function (13) since it only needs to be done when the scene loads. As before, we need to get a hold of the database with an `IDbConnection` (14) and create a new `IDbCommand` (15) to read the data. Since there is only one table and one value, it's quite simple for now. We can just read `all data` by using:  
+
+```sql
+"SELECT * FROM HitCountTableSimple"
+```
+
+In this case, `SELECT` stands for `read the following values`, followed by a `*` which indicates to read all the data. The keyword `FROM` then specifies the table that should be read from, which is again `HitCountTableSimple`. Finally, we execute this command using `ExecuteReader()` (17) since we expect data back. This data is saved in an [`IDataReader`](https://docs.microsoft.com/en-us/dotnet/api/system.data.idatareader?view=net-6.0), from the documentation:
+
+> Provides a means of reading one or more forward-only streams of result sets obtained by executing a command at a data source, and is implemented by .NET data providers that access relational databases.
+
+`IDataReader` addresses its content in an index fashion, where the ordering matches one of the columns in the SQL table. So in our case, `id` has index 0, and `hitCount` has index 1. The way this data is read is row by row. Each time we call `dataReader.Read()` (18), we read another row from the table. Since we know there is only one row in the table, we can just assign the `value` of that row to the `hitCount` using its index 1. The `value` is of type `INTEGER` so we need to use [`GetInt32(1)`](https://docs.microsoft.com/en-us/dotnet/api/system.data.idatarecord.getint32?view=net-6.0) to read it and specify the index of the field we want to read as a parameter, `id` being `0` and `value` being `1`.
```

Comment:
```suggestion
`IDataReader` addresses its content in an index fashion, where the ordering matches the one of the columns in the SQL table. So in our case, `id` has index 0, and `hitCount` has index 1. The way this data is read is row by row. Each time we call `dataReader.Read()` (18), we read another row from the table. Since we know there is only one row in the table, we can just assign the `value` of that row to the `hitCount` using its index 1. The `value` is of type `INTEGER` so we need to use [`GetInt32(1)`](https://docs.microsoft.com/en-us/dotnet/api/system.data.idatarecord.getint32?view=net-6.0) to read it and specify the index of the field we want to read as a parameter, `id` being `0` and `value` being `1`.
```